lib/mix/test/mix/tasks/compile.app_test.exs

```diff
@@ -60,49 +66,60 @@ defmodule Mix.Tasks.Compile.AppTest do
   test "generates .app file when changes happen" do
     Mix.Project.push(MixTest.Case.Sample)
 
-    in_fixture "no_mixfile", fn ->
+    in_fixture("no_mixfile", fn ->
       Mix.Tasks.Compile.Elixir.run([])
       assert Mix.Tasks.Compile.App.run([]) == :ok
 
-      contents = File.read!("_build/dev/lib/sample/ebin/sample.app")
-      assert contents =~ "{application,sample"
-      assert contents =~ "0.1.0"
-      assert contents =~ "'Elixir.A'"
-      assert contents =~ "{applications,[kernel,stdlib,elixir]}"
+      properties = parse_resource_file(:sample)
+      assert properties[:vsn] == '0.1.0'
+      assert properties[:modules] == [A, B]
+      assert properties[:applications] == [:kernel, :stdlib, :elixir]
 
       assert Mix.Tasks.Compile.App.run([]) == :noop
-    end
+    end)
   end
 
   test "uses custom application settings" do
     Mix.Project.push(CustomProject)
 
-    in_fixture "no_mixfile", fn ->
+    in_fixture("no_mixfile", fn ->
       Mix.Tasks.Compile.Elixir.run([])
       Mix.Tasks.Compile.App.run([])
-      contents = File.read!("_build/dev/lib/custom_project/ebin/custom_project.app")
-      assert contents =~ "0.2.0"
-      assert contents =~ "{maxT,infinity}"
-      assert contents =~ "{applications,[kernel,stdlib,elixir,logger,example_app]}"
-      assert contents =~ "Some UTF-8 description (uma descrição em UTF-8)"
-    end
+
+      properties = parse_resource_file(:custom_project)
+      assert properties[:vsn] == '0.2.0'
+      assert properties[:maxT] == :infinity
+      assert properties[:applications] == [:kernel, :stdlib, :elixir, :logger, :example_app]
+      assert properties[:description] == 'Some UTF-8 description (uma descrição em UTF-8)'
+    end)
   end
 
   test "automatically infers applications" do
     Mix.Project.push(CustomDeps)
 
-    in_fixture "no_mixfile", fn ->
+    in_fixture("no_mixfile", fn ->
       Mix.Tasks.Compile.Elixir.run([])
       Mix.Tasks.Compile.App.run([])
-      contents = File.read!("_build/dev/lib/custom_deps/ebin/custom_deps.app")
-      assert contents =~ "{applications,[kernel,stdlib,elixir,logger,ok1,ok3,ok4,ok7]}"
-    end
+
+      properties = parse_resource_file(:custom_deps)
+
+      assert properties[:applications] == [
+               :kernel,
+               :stdlib,
+               :elixir,
+               :logger,
+               :ok1,
+               :ok3,
+               :ok4,
+               :ok7
+             ]
```

Comment:
Please write this as:

```elixir
assert properties[:applications] ==
        [:kernel, :stdlib, :elixir, :logger, :ok1, :ok3, :ok4, :ok7]
```

Someetimes the formatter needs some hinting on which direction to take. :)